harness/determined/common/api/authentication.py

```diff
@@ -90,13 +94,22 @@ def _init_session(
             session_user = constants.DEFAULT_DETERMINED_USER
         assert session_user is not None
 
+        # Check the token store to determine if session_user has valid login credentials.
+        # If there are no such credentials, token will be set to (or will remain) None.
```

Comment:
Check the token store if this session_user has a cached token.  If so, check with the master to verify it has not expired.  Otherwise, let the token be None.